Fix payment in connect mode

ptibogxiv opened the pull request:
I know that it's not an optimal fix but it's functionnal for dolibarr 8.0.0 release

ptibogxiv committed:
Fix payment in connect mode

I know that it's not an optimal fix but it's functionnal for dolibarr 8.0.0 release

ptibogxiv committed:
Fix and compatibility for V8  and connect

fix for V8 release and optimal functionnality with connect mode

ptibogxiv committed:
add statement descriptor as newpayment.php

ptibogxiv committed:
fix spaces

eldy commented on htdocs/stripe/class/stripe.class.php:
```diff
@@ -367,18 +367,20 @@ public function createPaymentStripe($amount, $currency, $origin, $item, $source,
 			"dol_id" => "" . $item . "",
 			"dol_type" => "" . $origin . "",
 			"dol_thirdparty_id" => "" . $societe->id . "",
+      "FULLTAG" => $description,
```
Why adding FULLTAG ? We already have "dol_type" (saying it is order or invoice) and "dol_id" giving id of order or invoice.
FULLTAG was introduced for payment mode that manage only 1 param to be able to provide several data in 1 string, but with stripe, we already have structured data that we can manage.

ptibogxiv replied:
because page with list of payment use it but we can change this other page and pass.

eldy committed:
Update stripe.class.php

eldy commented on htdocs/stripe/class/stripe.class.php:
```diff
@@ -396,16 +398,19 @@ public function createPaymentStripe($amount, $currency, $origin, $item, $source,
 						"currency" => "$currency",
-						// "statement_descriptor" => " ",
+                                                "capture"  => true,
+						"statement_descriptor" => dol_trunc(dol_trunc(dol_string_unaccent($customer->name), 6, 'right', 'UTF-8', 1).' '.$description, 22, 'right', 'UTF-8', 1),     // 22 chars that appears on bank receipt
```
It is recommended by Stripe to add the seller business name so customer know who he paid. Otherwise, customer has no chance to know which companies he paid and he complain to its bank to be refunded, and this make Stripe to not refund moneys to seller.

So i suggest to put instead 
"statement_descriptor" => dol_trunc(dol_trunc(dol_string_unaccent($mysoc->name), 6, 'right', 'UTF-8', 1).' '.$description, 22, 'right', 'UTF-8', 1)

ptibogxiv replied:
done ;-)

eldy commented on htdocs/stripe/class/stripe.class.php:
```diff
@@ -425,19 +430,26 @@ public function createPaymentStripe($amount, $currency, $origin, $item, $source,
+						"source" => "$source",
+						"customer" => "$customer"
+					);
+					if ($conf->entity>1 && $fee>0)
```
Why using conf->entity > 1 ?
When using multicompany, company 1 is a company like other so like the company 2 or 3, so we should not have code that differs if we are company 1 or 2.

ptibogxiv replied:
I define a new constant for define principal entity (principal entity can't have application fee without error of API)

ptibogxiv committed:
Fix statement descriptor $soc->name

ptibogxiv committed:
Define introduce principal entity

define which entity must be use as principal with no fee and everything else (cf stripe API)

eldy committed:
Update stripe.class.php

eldy committed:
Update stripe.class.php

eldy committed:
Update stripe.class.php
